src/txorphanpool.cpp

```diff
@@ -40,10 +40,7 @@ bool CTxOrphanPool::AddOrphanTx(const CTransaction &tx, NodeId peer)
     }
 
     uint64_t txSize = RecursiveDynamicUsage(tx);
-    mapOrphanTransactions[hash].ptx = MakeTransactionRef(tx);
-    mapOrphanTransactions[hash].fromPeer = peer;
-    mapOrphanTransactions[hash].nEntryTime = GetTime();
-    mapOrphanTransactions[hash].nOrphanTxSize = txSize;
+    mapOrphanTransactions.emplace(hash, COrphanTx{MakeTransactionRef(tx), peer, GetTime(), txSize});
```

Comment:
somewhere else in the PR `sizeof(ptx)` is used rather than `RecursiveDynamicUsage(tx)`, is this a problem here?

Reply:
thanks for pointing this out, the code for orhapntx is correct however for thinkblocks i need to still use the size of the tx rather than the size of the smart pointer.

Reply:
Actually, I've gotten a bit muddled...the new code is correct as are the tests... The reason why we only have to count the size of the shared pointer now when we reconstruct the block is because the txns are already in memory...they are either in the mempoool, orphanpool, or mapmissingtx.  As such we don't have to count their bytes again, but only the size of the shared pointer, because a CBlock is now a header plus shared pointers to txns, rather than a header plus the raw txns...  However, for the orphanpool, as you noticed, we still need the actual size of the txn...even though we use a shared pointer, the txn still takes up memory when it first arrives in orphan pool.  To be accurate though we should now count the size of the pointer plus the txn when calculating the size of the orphan pool bytes.

Reply:
thanks for the explanation Peter!